cli/flox-rust-sdk/src/models/environment/core_environment.rs

```diff
@@ -116,14 +231,20 @@ impl<State> CoreEnvironment<State> {
     /// It's included in the [ReadOnly] struct for ergonomic reasons
     /// and because it doesn't modify the manifest.
     ///
-    /// todo: should we always write the lockfile to disk?
+    /// The caller is responsible for skipping calls to lock when an environment
+    /// is already locked.
+    /// For that reason, this always writes the lockfile to disk.
     pub fn lock(&mut self, flox: &Flox) -> Result<LockedManifest, CoreEnvironmentError> {
         let manifest = self.manifest()?;
 
         let lockfile = match manifest {
             TypedManifest::Pkgdb(_) => {
-                tracing::debug!("using pkgdb to lock");
-                LockedManifest::Pkgdb(self.lock_with_pkgdb(flox)?)
+                if *flox.features.use_catalog {
+                    return Err(CoreEnvironmentError::LockingVersion0NotSupported);
+                } else {
+                    tracing::debug!("using pkgdb to lock");
+                    LockedManifest::Pkgdb(self.lock_with_pkgdb(flox)?)
```

Comment:
Can you describe why we're keeping this code path if we no longer lock v0? Is it just to support our existing integration tests, in which case are those tests still valid, or is it to support user-facing behaviour?

Reply:
I already had to fight a lot of test breakage, so I didn't want to break anything that sets `FLOX_FEATURES_USE_CATALOG=false`. That said I didn't even check if it would break much. And I was using `FLOX_FEATURES_USE_CATALOG=false` to generate some of the data for test cases, so it was partially for my own convenience

I'm expecting we'll drop it next

Reply:
Thanks 👍🏻 Then it's correct to say from a product/user perspective that we aren't supporting the use of `FLOX_FEATURES_USE_CATALOG=false` and that we can (after this PR) remove those duplicate integration tests?

Reply:
Yep. My understanding is we still need to support existing v0 manifests, so we may need to use the pre-generated v0 manifests in `test_data` to keep some tests around. But we should be able to set `FLOX_FEATURES_USE_CATALOG=true` everywhere